src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java

```diff
@@ -1,47 +1,151 @@
+  private static final String CONCEPT_NAME = "concept_name";
+  private static final String ANCESTOR_CONCEPT_ID = "ancestor_concept_id";
+  private static final String DESCENDANT_CONCEPT_ID = "descendant_concept_id";
+
+  /**
+   * Generate a query that retrieves the children of the given concept and their roll-up counts.
+   *
+   * <p>SELECT cc.concept_id, cc.concept_name, COUNT(DISTINCT co.person_id) as count FROM `concept`
+   * AS cc JOIN `concept_ancestor` AS ca ON cc.concept_id = ca.ancestor_concept_id JOIN
+   * `'domain'_occurrence` AS co ON co.'domain'_concept_id = ca.descendant_concept_id WHERE
+   * (c.concept_id IN (SELECT c.descendant_concept_id FROM `concept_ancestor` AS c WHERE
+   * c.ancestor_concept_id = conceptId) AND c.concept_id != conceptId) GROUP BY cc.concept_id,
+   * cc.concept_name ORDER BY cc.concept_name ASC
+   */
   public static String buildConceptChildrenQuery(
-      int conceptId, TableNameGenerator tableNameGenerator, CloudPlatformWrapper platform) {
-    TablePointer conceptTablePointer = TablePointer.fromTableName("concept", tableNameGenerator);
-    TableVariable conceptTableVariable = TableVariable.forPrimary(conceptTablePointer);
-    FieldVariable nameFieldVariable = conceptTableVariable.makeFieldVariable("concept_name");
-    FieldVariable idFieldVariable = conceptTableVariable.makeFieldVariable("concept_id");
+      SnapshotBuilderDomainOption domainOption,
+      int conceptId,
+      TableNameGenerator tableNameGenerator,
+      CloudPlatformWrapper platform) {
+
+    // concept table and its fields concept_name and concept_id
+    var conceptTablePointer = TablePointer.fromTableName(CONCEPT, tableNameGenerator);
+    var conceptTableVariable = TableVariable.forPrimary(conceptTablePointer);
+    var nameFieldVariable = conceptTableVariable.makeFieldVariable(CONCEPT_NAME);
+    var idFieldVariable = conceptTableVariable.makeFieldVariable(CONCEPT_ID);
+
+    // concept_ancestor joined on concept.concept_id = ancestor_concept_id
+    var conceptAncestorTablePointer =
+        TablePointer.fromTableName(CONCEPT_ANCESTOR, tableNameGenerator);
+    var conceptAncestorTableVariable =
+        TableVariable.forJoined(conceptAncestorTablePointer, ANCESTOR_CONCEPT_ID, idFieldVariable);
+    var descendantIdFieldVariable =
+        conceptAncestorTableVariable.makeFieldVariable(DESCENDANT_CONCEPT_ID);
+
+    // domain specific occurrence table joined on concept_ancestor.descendant_concept_id =
+    // 'domain'_concept_id
+    var domainOccurrenceTablePointer =
+        TablePointer.fromTableName(domainOption.getTableName(), tableNameGenerator);
+    var domainOccurenceTableVariable =
+        TableVariable.forJoined(
+            domainOccurrenceTablePointer, domainOption.getColumnName(), descendantIdFieldVariable);
+
+    // COUNT(DISTINCT person_id)
+    var countFieldVariable =
+        new FieldVariable(
+            new FieldPointer(domainOccurrenceTablePointer, PERSON_ID, "COUNT"),
```

Comment:
To avoid having to keep the tablepointer around here, you could add an overload to `TableVariable.makeFieldVariable()` that accepts an `alias` parameter.